openvmm/hvlite_core/src/worker/vm_loaders/igvm.rs

```diff
@@ -253,7 +254,8 @@ pub fn vtl2_memory_range(
     let physical_address_size = physical_address_size - 1;
 
     // Create an initial memory layout to determine the highest used address.
-    let dummy_layout = MemoryLayout::new(mem_size, mmio_gaps, None).map_err(Error::MemoryConfig)?;
+    let dummy_layout = MemoryLayout::new(mem_size, mmio_gaps, device_reserved_gaps, None)
```

Comment:
I can't leave a comment, but i think the `end_of_ram_or_mmio` fn needs to change to just be `end_of_layout`? since now we have 3 different range types.